### PR TITLE
add permisions precondition

### LINKED ISSUE #15982: Permission preconditions

### PR DESCRIPTION
**Pull request by Geometer1729**
Explain your changes:
* Adds a permissions field to the account update preconditions

Explain how you tested your changes:
* currently only limited testing through o1js

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #15982 (with https://github.com/o1-labs/o1js/pull/1862 and https://github.com/o1-labs/o1js-bindings/pull/306)

### CODE REVIEW

**Comment by Geometer1729** [comment]
!ci-nix-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

File: src/lib/mina_base/test/zero_vesting_period.ml [hunk 1]
```diff
@@ -138,6 +138,21 @@ let mk_zkapp_with_vesting_period n =
...
+                "access": [ "Ignore" ],
+                "set_delegate": [ "Ignore" ],
+                "set_permissions": [ "Ignore" ],
+                "set_verification_key": [ "Ignore" ],
+                "set_zkapp_uri": [ "Ignore" ],
+                "edit_action_state": [ "Ignore" ],
+                "set_token_symbol": [ "Ignore" ],
+                "increment_nonce": [ "Ignore" ],
+                "set_voting_for": [ "Ignore" ],
+                "set_timing": [ "Ignore" ],
```
Comment (mrmr1993) [inline src/lib/mina_base/test/zero_vesting_period.ml:154]: This trailing comma is not valid JSON.

File: src/lib/mina_base/test/zero_vesting_period.ml [hunk 2]
```diff
@@ -246,6 +261,21 @@ let mk_zkapp_with_vesting_period n =
...
```
Comment (mrmr1993) [inline src/lib/mina_base/test/zero_vesting_period.ml:277]: Similarly here

File: src/lib/mina_base/test/zero_vesting_period.ml [hunk 3]
```diff
@@ -138,6 +138,21 @@ let mk_zkapp_with_vesting_period n =
                 "proved_state": [ "Ignore" ],
                 "is_new": [ "Ignore" ]
               },
+              "permissions": {
```
Comment (mrmr1993) [inline src/lib/mina_base/test/zero_vesting_period.ml:141]: This also seems to be in the wrong place, it should be in `account` above.

File: src/lib/mina_base/test/zero_vesting_period.ml [hunk 4]
```diff
@@ -246,6 +261,21 @@ let mk_zkapp_with_vesting_period n =
...
```
Comment (mrmr1993) [inline src/lib/mina_base/test/zero_vesting_period.ml:264]: Ditto here

File: src/lib/mina_base/test/zero_vesting_period.ml (hunk 3 above)
Comment (Geometer1729) [reply src/lib/mina_base/test/zero_vesting_period.ml:141]: Thanks for catching that, I moved the permissions inside account field elsewhere but forgot to move it here. It seems like that was causing the flaky proof cache issue.

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
sorry closed the wrong pr

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Trivo25** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Trivo25** [comment]
!ci-build-me

**Comment by Trivo25** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Pull request closed without merging**